File: linearQueueLinkedList/queue.cpp
#include "node.cpp"
using namespace std;
class Queue
{
	private:
		Node *front, *rear;
	public:
		Queue()
		{
			front=rear=NULL;
		}
		Queue(Node *front,Node *rear)
		{
			this->front=front;
			this->rear=rear;
		}
		bool isEmpty()
		{
			return front==NULL;
		}
		void enqueue()
		{
			Node *nptr=new Node();
			if(nptr==NULL)
				cout<<"\nOverflow"<<endl;
			else
			{
				cout<<"\nEnter item: ";
				int item;
				cin>>item;
				nptr->setData(item);
				if(isEmpty())
					front=rear=nptr;
				else
				{
					rear->setLink(nptr);
					rear=rear->getLink();
				}
				show();
			}
		}
		void dequeue()
		{
			if(isEmpty())
				cout<<"\nUnderflow"<<endl;
			else
			{
				cout<<"\n"<<front->getData()<<endl;
				Node *temp=front;
				if(front==rear)
					front=rear=NULL;
				else
					front=front->getLink();
				delete temp;
				temp=NULL;
				show();
			}
		}
		void show()
		{
			if(isEmpty())
				cout<<"\nEmpty"<<endl;
			else
			{
				cout<<"\nQueue"<<endl;
				Node *ptr=front;
				while(ptr!=NULL)
				{
					cout<<ptr->getData()<<"\t";
					ptr=ptr->getLink();
				}
				cout<<endl;
			}
		}
		~Queue()
		{
			while(front!=NULL)
			{
				Node *temp=front;
				front=front->getLink();
				delete temp;
				temp=NULL;
			}
			rear=NULL;
		}
};

File: linearQueueLinkedList/Queue.java
import java.util.Scanner;
class Queue
{
    private Node front,rear;
    Queue()
    {
        front=null;
        rear=null;
    }

    Queue(Node front,Node rear)
    {
        this.front=front;
        this.rear=rear;
    }

    void enqueue()
    {
        Scanner sc=new Scanner(System.in);
        Node nptr=new Node();
        if(nptr==null)
            System.out.println("\nOverflow");
        else
        {
            System.out.print("\nEnter item: ");
            nptr.setData(sc.nextInt());
            if(isEmpty())
            {
                front=nptr;
                rear=nptr;
            }
            else
            {
                rear.setLink(nptr);
                rear=nptr;
            }
            show();
        }
    }

    boolean isEmpty()
    {
        return front==null;
    }

    void show()
    {
        if(isEmpty())
            System.out.println("\nEmpty");
        else
        {
            System.out.println("\nQueue");
            Node ptr=front;
            while(ptr!=null)
            {
                System.out.print(ptr.getData()+"\t");
                ptr=ptr.getLink();
            }
            System.out.println();
        }
    }

    void dequeue()
    {
        if(isEmpty())
            System.out.println("\nUnderflow");
        else
        {
            System.out.println("\n"+front.getData());
            if(front==rear)
            {
                front=null;
                rear=null;
            }
            else
                front=front.getLink();
            show();
        }
    }
}